app/harvesters/nypl_harvester.rb

```diff
@@ -0,0 +1,238 @@
+##
+# A harvester for NYPL's API
+#
+class NyplHarvester
+  include Krikri::Harvester
+
+  DEFAULT_URI = 'http://api.repo.nypl.org/api/v1'
+  DEFAULT_NAME = 'nypl'
+  DEFAULT_BATCHSIZE = 10
+  DEFAULT_THREAD_COUNT = 5
+
+  ##
+  # Initialize, and set default options as appropriate.
+  #
+  # @param opts [Hash] a hash of options as defined by {.expected_opts}
+  #
+  # @example
+  #    Typical instantiation, good for most cases:
+  #
+  #        NyplHarvester.new(nypl: { apikey: '[key]' })
+  #
+  # Parameters to 'opts':
+  #
+  # - uri:        See Krikri::Harvester#initialize.
+  #               Defaults to "http://api.repo.nypl.org/api/v1'"
+  # - name:       See Krikri::Harvester#initialize.  Defaults to "nypl"
+  # - batchsize:  The number of records to fetch with each API request.
+  #               Defaults to 10.
+  # - threads:    The number of record pages to fetch in parallel.
+  #               Defaults to 5.
+  #
```

Comment:
It would be good if the `apikey` parameter referenced above in the example had a line somewhere around here.